rsocket-core/src/main/java/io/rsocket/lease/LeaseEventsListener.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright 2015-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.rsocket.lease;
+
+public interface LeaseEventsListener {
+
+  void onEvent(Lease lease, Type eventType);
```

Comment:
@robertroeser for stats living externally

Reply:
The stats object can live external and then counter an be updated. I don't think we should be passing around an object on every call when leases is called.